tasks/main.yml

```diff
@@ -1,6 +1,11 @@
 ---
-- name: add the OS specific variables
-  include_vars: '{{ ansible_os_family }}.yml'
+- name: Set OS dependent variables
+  include_vars: "{{ item }}"
+  with_first_found:
+   - '{{ ansible_distribution }}-{{ ansible_distribution_major_version }}.yml'
+   - '{{ ansible_distribution }}.yml'
+   - '{{ ansible_os_family }}-{{ ansible_distribution_major_version }}.yml'
+   - '{{ ansible_os_family }}.yml'
```

Comment:
Something like this:

```YAML
- name: Set OS family dependent variables
  include_vars: '{{ ansible_os_family }}.yml'

- name: Set OS dependent variables
  include_vars: '{{ item }}'
  with_first_found:
   - '{{ ansible_distribution }}-{{ ansible_distribution_major_version }}.yml'
   - '{{ ansible_distribution }}.yml   - '{{ ansible_os_family }}-{{ ansible_distribution_major_version }}.yml'
```

Then `{{ ansible_os_family }}.yml` could contain all the basics and other files could be kept at a minimum.